### PR TITLE
azure: add option to enable boot diagnostics for VMs in azure

### PR DESCRIPTION
**Pull request by cpanato**
in order to collect the boot logs for debug investigation. we need to enable the boot diagnostics on Azure.

fixes: INST-904

next step is to start collecting the boot logs in the test framework. This PR enable this feature in the tests area.

### CODE REVIEW

File: platforms/azure/variables.tf [hunk 1]
```diff
@@ -294,3 +294,9 @@ EOF
     usdodeast          = 2
   }
 }
+
+variable "tectonic_azure_enable_boot_diagnostics" {
```
Comment (sym3tri) [inline platforms/azure/variables.tf]: If there are similar settings for other cloud providers it's probably better to have a single top-level variable that each of them is aware of. You can indicate which platforms are supported in the var description.

Comment (cpanato) [reply platforms/azure/variables.tf]: I will make top level. thanks for the review

Comment (cpanato) [reply platforms/azure/variables.tf]: done @sym3tri

**Comment by cpanato** [comment]
retest this please

**Comment by cpanato** [comment]
@squat  / @alexsomesan  please review when you guys have time. thanks

**Comment by sym3tri** [comment]
thanks for making the changes. seems good to me, but i'll let @alexsomesan give final review

**Comment by justaugustus** [comment]
+1

File: modules/azure/resource-group/resource-group.tf [hunk 2]
```diff
@@ -35,10 +39,30 @@ resource "azurerm_resource_group" "tectonic_cluster" {
...
+
 output "name" {
   value = "${var.external_rsg_id == "" ? element(concat(azurerm_resource_group.tectonic_cluster.*.name, list("")), 0) : element(split("/", var.external_rsg_id), 4)}"
 }
 
 output "storage_id" {
   value = "${random_id.storage_id.hex}"
 }
+
+output "storage_blob_endpoint" {
```
Comment (squat) [inline modules/azure/resource-group/resource-group.tf]: we should break apart this module into variables.tf and outputs.tf to match the other modules

Comment (cpanato) [reply modules/azure/resource-group/resource-group.tf]: done

**Pull request merged**